homeassistant/components/mqtt/vacuum/__init__.py

```diff
@@ -100,7 +112,7 @@ async def _async_setup_entity(
     discovery_data: dict | None = None,
 ) -> None:
     """Set up the MQTT vacuum."""
-    setup_entity = {
+    setup_entity: dict[str, Callable[..., Coroutine[Any, Any, None]]] = {
```

Comment:
Same here, let it be inferred
```suggestion
    setup_entity = {
```